### PR TITLE
Select date on IOS UIDatePicker with actionForSetDate

### PR DESCRIPTION
**Pull request by matthewrfindley**
- [x] This is a small change 

---

**Description:**

Addresses the following issues:
https://github.com/wix/Detox/issues/778
https://github.com/wix/Detox/issues/1051

## The problem:
Using `setColumnToValue` to select the correct date on a DatePickerIOS does not work as expected.  This [issue](https://github.com/wix/Detox/issues/1051) defines a solution to select a valid value from the `0` column by using the value of the label.  The solution breaks down because the date picker contains the same label for every year in the list.  Attempting to select `Feb 8` for 2019 would result in selecting `Feb 8` for 2006, which is the minimum date in the list.

Example usage of setColumnToValue on DatePickerIOS:
```
await element(by.type('UIPickerView')).setColumnToValue(0,"Feb 6");
```

react-native DatePickerIOS renders the date in the UIDatePicker as 2 labels.  The first being the 3 letter month abbreviation and day, the second being the 3 letter day abbreviation.  Using `setColumnToValue` breaks down when trying to select the correct date since the implementation of Earl Grey's `GREYPickerAction` selects only the first label see [here](https://github.com/google/EarlGrey/blob/master/EarlGrey/Action/GREYPickerAction.m#L97).  Even if Earl Grey were able to compare using all the labels with a list item, a UIDatePicker would be limited to only 7 years worth of dates.  This would not be maintainable since we are not in control of the default range for UIDatePicker, nor is it acceptable to force users to limit the list in their application.

## Solution:
Expose an existing Earl Grey action called [actionForSetDate](https://github.com/google/EarlGrey/blob/master/EarlGrey/Action/GREYActions.m#L272) within Detox.

### CODE REVIEW

**Comment by LeoNatan** [comment]
Thanks for submitting this PR. Will review shortly.

File: detox/ios/Detox/GREYActions+Detox.h [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+//  Created by Matt Findley on 2/7/19.
+//  Copyright © 2019 Wix. All rights reserved.
+//
+
+@import Foundation;
+#import <EarlGrey/EarlGrey.h>
+
+@interface GREYActions (Detox)
+
++ (id<GREYAction>)detoxSetDatePickerDateIOSOnly:(NSString *)dateString withFormat:(NSString *)dateFormat;
```
Comment (LeoNatan) [inline detox/ios/Detox/GREYActions+Detox.h]: Let's not call this "IOSOnly". Should be `detoxSetDatePickerDate:withFormat:`.

File: detox/ios/Detox/GREYActions+Detox.m [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+//  Copyright © 2019 Wix. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "GREYActions+Detox.h"
+#import <EarlGrey/GREYActions.h>
+
+@implementation GREYActions (Detox)
+
++ (id<GREYAction>)detoxSetDatePickerDateIOSOnly:(NSString *)dateString withFormat:(NSString *)dateFormat
```
Comment (LeoNatan) [inline detox/ios/Detox/GREYActions+Detox.m]: Likewise, `detoxSetDatePickerDate:withFormat:`.

File: detox/test/e2e/17.datePicker.test.js [hunk 3]
```diff
@@ -3,12 +3,17 @@ describe(':ios: DatePicker', () => {
...
     it('datePicker should trigger change handler correctly', async () => {
       await element(by.type('UIPickerView')).setColumnToValue(1,"6");
       await element(by.type('UIPickerView')).setColumnToValue(2,"34");
       await expect(element(by.id("timeLabel"))).toHaveText('choosenTime is 6:34');
     });
 
-  });
-  
+    it('can select dates on a UIDatePicker', async () => {
+      await element(by.type('UIDatePicker')).setDatePickerDateIOSOnly('2019-02-06T05:10:00-08:00', "yyyy-MM-dd'T'HH:mm:ssZZZZZ");
```
Comment (LeoNatan) [inline detox/test/e2e/17.datePicker.test.js]: This should be `setDatePickerDate` and documentation should be updated to state it is for iOS only, as are the current picker view APIs.

File: detox/src/ios/expect.js [hunk 4]
```diff
@@ -185,6 +186,14 @@ class ScrollColumnToValue extends Action {
   constructor(column,value) {
     super();
     this._call = invoke.callDirectly(GreyActions.actionForSetPickerColumnToValue(column,value))
+    this._call = invoke.callDirectly(GreyActions.actionForSetPickerColumnToValue(column, value));
```
Comment (LeoNatan) [inline detox/src/ios/expect.js]: I think this line was duplicated by mistake?

Comment (matthewrfindley) [reply detox/src/ios/expect.js]: 📗 ` [fe3a615](/wix/Detox/pull/1148/commits/fe3a6152e98f1d8fe352c78fde9ded4fe7d2168e)`

File: detox/test/e2e/17.datePicker.test.js (hunk 3 above)
Comment (matthewrfindley) [reply detox/test/e2e/17.datePicker.test.js]: 📗 9470d65

File: detox/ios/Detox/GREYActions+Detox.m (hunk 2 above)
Comment (matthewrfindley) [reply detox/ios/Detox/GREYActions+Detox.m]: 📗 1d6913c

File: detox/ios/Detox/GREYActions+Detox.h (hunk 1 above)
Comment (matthewrfindley) [reply detox/ios/Detox/GREYActions+Detox.h]: 📗 1d6913c

**Comment by matthewrfindley** [comment]
@LeoNatan https://jenkins-oss.wixpress.com/job/multi-detox-pr/920/ - Failure seems unrelated to any of the recent changes. 🤷‍♂️

**Comment by LeoNatan** [comment]
Looks good. CI is flaky.

**Comment by LeoNatan** [comment]
@rotemmiz Let's merge?

**Comment by rotemmiz** [comment]
Android e2e hangs, wonder why

File: detox/test/src/Screens/DatePickerScreen.js [hunk 5]
```diff
@@ -1,47 +1,49 @@
 import React, { Component } from 'react';
-import {
-  Text,
-  View,
-  StyleSheet,
-  DatePickerIOS
-} from 'react-native';
+import { Text, View, StyleSheet, DatePickerIOS } from 'react-native';
```
Comment (rotemmiz) [inline detox/test/src/Screens/DatePickerScreen.js]: You can't add `DatePickerIOS` on a shared component, it needs to run on both iOS and Android.
You might need to split this screen to .ios.js and .android.js.
This is probably the reason why the e2e hangs, the app crashes.

**Comment by rotemmiz** [comment]
Please see my comment, I think I found the reason for Android E2E's hangs

File: detox/test/src/Screens/DatePickerScreen.js (hunk 5 above)
Comment (LeoNatan) [reply detox/test/src/Screens/DatePickerScreen.js]: But that’s not a new change. How did it work until now?

Comment (rotemmiz) [reply detox/test/src/Screens/DatePickerScreen.js]: ho, you're right, I haven't noticed that

**Pull request merged**